Game/ui/inventorymenu.cpp

```diff
@@ -308,34 +330,59 @@ InventoryMenu::PageLocal &InventoryMenu::activePageSel() {
   return pageLocal[1];
   }
 
-void InventoryMenu::onTakeStuff() {
+void InventoryMenu::onTakeStuff() { 
+  uint8_t itemCount = 0;
   auto& page = activePage();
-  auto& sel  = activePageSel();
-  if(sel.sel>=page.size())
+  auto& sel = activePageSel();
+  if(sel.sel >= page.size())
     return;
   auto& r = page[sel.sel];
+  if(r.isEquiped()) {
```

Comment:
This check won't work as expected, if r.count()>1.

Reply:
Yeah.. It can move all the stack until r.count()==1, so it doesn't touch _the one_ equipped. For sure the best would be to separate them in two stacks but don't know how.